src/test/java/org/apache/sysds/test/functions/builtin/part2/BuiltinOutlierByDBTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysds.test.functions.builtin.part2;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import org.apache.sysds.common.Types.ExecMode;
+import org.apache.sysds.common.Types.ExecType;
+import org.apache.sysds.runtime.matrix.data.MatrixValue.CellIndex;
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Test;
+import java.util.HashMap;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Collection;
+
+public class BuiltinOutlierByDBTest extends AutomatedTestBase
+{
+	private final static String TEST_NAME = "outlierByDB";
+	private final static String TEST_DIR = "functions/builtin/";
+	private static final String TEST_CLASS_DIR = TEST_DIR + BuiltinOutlierByDBTest.class.getSimpleName() + "/";
+
+	private final static double eps = 1e-3;
```

Comment:
aim for a good `eps`, perhaps `1e-9`. 🎯

Reply:
The issue is the tests eps is either 0.0 or 1.0, since I return a yes/no mapping of outliers. 
Whats confusing me is that I get the exact same points, with some seemingly random ones added.

Reply:
I will take a closer look now. :)

Reply:
I figured something out: if I do some preprocessing in R. Removing the outliers and calling dbscan again, all the tests work 🤔